libhwmon/wraphelper.h

```diff
@@ -29,3 +29,26 @@ static int wrap_dlclose(void *h) {
 	return dlclose(h);
 }
 #endif
+//Converts from 8 fractional bit int to double (ADL method result format)
+static double adlf2double(int adlf){
+	double res = 0.0;
+	int fraction = 	adlf & 0x00FF; //last 8 bits seem to be decimal part
+	int number = 	(adlf & 0xFF00) >> 8; //first 8 bits are the integer part
+	double dfraction = 0; //decimal part
+	int i = 1;
+	while(fraction > 0){
```

Comment:
You don't need to do all of this crazy stuff to convert fractional part.

Just do: return double(adlf << 8) / 256.0;

I don't think you do the fractional part correctly. It doesn't go 0..99, it goes 0..255

Reply:
In fact, I'd get rid of this whole function and just do the conversion wherever it's called. Didn't check but it's probably just one or 2 places.

Reply:
Yeah, I guess you're totally right! I'm about to correct

Reply:
Shouldn't it be 
return double(adlf & 0x00FF) / 256 ?

Reply:
Ints with fractional parts are just normal ints where bit 0 represents 2^-8 instead of 2^0. So you can simply convert with double(adlf * 256) / 256.0.

bit 0 - 1/256 watts
bit 1 - 1/128 watts
...
bit 7 - 1/2 watt
bit 8 - 1 watt
bit 9 2 watts
and so on...

You don't treat the integer and fraction separately, it's just one number.

Reply:
Err... my apollogies. You're right. I it should be simply double(adlf) / 256.0;